src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -63,22 +63,21 @@ export default abstract class QueryBase implements QueryInterface {
     deposit: Deposit,
     amountToRelay: BigNumberish,
     relayAddress = DEFAULT_SIMULATED_RELAYER_ADDRESS
-  ): Promise<BigNumberish> {
+  ): Promise<{ gasCost: BigNumberish; gasTokenCost: BigNumberish }> {
     const relayerToSimulate = relayAddress ?? this.simulatedRelayerAddress;
     const tx = await createUnsignedFillRelayTransactionFromDeposit(
       this.spokePool,
       deposit,
       toBN(amountToRelay),
       relayerToSimulate
     );
-    const estimatedGas = await estimateTotalGasRequiredByUnsignedTransaction(
+    return await estimateTotalGasRequiredByUnsignedTransaction(
```

Comment:
Any reason for this change other than style? I agree with it, and we could go one step further and remove the `await`
```suggestion
    return estimateTotalGasRequiredByUnsignedTransaction(
```